src/graph/SchemaHelper.cpp

```diff
@@ -246,5 +251,42 @@ Status SchemaHelper::alterSchema(const std::vector<AlterSchemaOptItem*>& schemaO
     return Status::OK();
 }
 
+StatusOr<int64_t> SchemaHelper::toTimestamp(const VariantType &value) {
+    if (value.which() != VAR_INT64 && value.which() != VAR_STR) {
+        return Status::Error("Invalid value type");
+    }
+
+    int64_t timestamp;
+    if (value.which() == VAR_STR) {
+        static const std::regex reg("^([1-9]\\d{3})-"
+                                    "(0[1-9]|1[0-2]|\\d)-"
+                                    "(0[1-9]|[1-2][0-9]|3[0-1]|\\d)\\s+"
+                                    "(20|21|22|23|[0-1]\\d|\\d):"
+                                    "([0-5]\\d|\\d):"
+                                    "([0-5]\\d|\\d)$");
+        std::smatch result;
+        if (!std::regex_match(boost::get<std::string>(value), result, reg)) {
+            return Status::Error("Invalid timestamp type");
+        }
+        struct tm time;
+        memset(&time, 0, sizeof(time));
+        time.tm_year = atoi(result[1].str().c_str()) - 1900;
+        time.tm_mon = atoi(result[2].str().c_str()) - 1;
+        time.tm_mday = atoi(result[3].str().c_str());
+        time.tm_hour = atoi(result[4].str().c_str());
+        time.tm_min = atoi(result[5].str().c_str());
+        time.tm_sec = atoi(result[6].str().c_str());
```

Comment:
I think it's best to add a `TODO `here
Time type validity check, such as `2020-02-29` or` 2020-04-31`

Reply:
Good catch, pr #1450 has done it.